### PR TITLE
Convert Rest of Ole to Cswin32

### PR DESCRIPTION
**Pull request by lonitra**
- Adds test to capture our wrapping behavior:  
    - Data coming in must implement `System.Runtime.InteropServices.ComTypes.IDataObject` to be taken as is. If it does not, we will wrap the data in a `DataObject`
    - Data going out must implement `System.Windows.Forms.IDataObject` to be passed out as is. If it does not, we will wrap the data in a `DataObject`
- Convert the remaining ole functions to Cswin32
- Other minor clean ups
- Add test capturing clipboard proxy behavior

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10651)

### CODE REVIEW

**Comment by elachlan** [comment]
@kant2002 are you still interested in Winforms Clipboard interop for AOT?

**Comment by kant2002** [comment]
I think that this is valuable, but i do not have capacity for this work unfortunately. Anything else I can help with?

**Comment by elachlan** [comment]
Just keeping you informed.

**Comment by willibrandon** [comment]
I have fetched your branch and initial testing of drag and drop looks good to me.

**Comment by lonitra** [comment]
> I have fetched your branch and initial testing of drag and drop looks good to me.

Thanks! Appreciate the extra eyes (:

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs [hunk 1]
```diff
@@ -5198,12 +5198,44 @@ public DragDropEffects DoDragDrop(object data, DragDropEffects allowedEffects)
     ///   value.
     ///  </para>
     /// </remarks>
-    public DragDropEffects DoDragDrop(
+    public unsafe DragDropEffects DoDragDrop(
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:5201]: nit: the Control is `unsafe` this isn't necessary

**Review by JeremyKuhne — APPROVED** [review]
Looks great!

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs (hunk 1 above)
Comment (lonitra) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:5201]: Ah nice catch! I'll address this in a follow up

**Pull request merged**